### DISCUSSION TITLE
LottieLogger.assertionFailure/Swift.assertionFailure Crash when asset is not found in bundle.

### DISCUSSION THREAD

**Discussion by rfigini** [airbnb/lottie-ios#1677] [General]
## Which Version of Lottie are you using?

Lottie 3.4.1

## Expected Behavior 
Not crash when execute AnimationImageProvider.imageForAsset and not found the asset in bundle.

## Actual Behavior
Crash when execute AnimationImageProvider.imageForAsset and not found the asset in bundle.
![image](https://user-images.githubusercontent.com/27892468/181764132-099c354f-c414-49ba-8b59-18493fc0b420.png)
![image](https://user-images.githubusercontent.com/27892468/181764186-847640e3-3763-402f-802e-614e1fe5d04a.png)
![image](https://user-images.githubusercontent.com/27892468/181764223-f0716527-3328-412f-adfc-ed5e7b6d0a44.png)
![image](https://user-images.githubusercontent.com/27892468/181764252-b5d701a8-2968-4e5f-bb5e-fc9bb0099427.png)

I guess the library wants to log an exception but it crashes, this doesn't happen at ver. 3.3.0
Could you help me?
Thanks !

**Comment by calda** [comment]
`Swift.assertionFailure` crashes in debug builds but is a no-op in production builds, so this wouldn't happen in production.

If you prefer not to fix the underlying issue where Lottie cannot find the image in your animation json file, you can customize the `LottieLogger.shared.assertionFailure` method to not call `Swift.assertionFailure`. For example you can do:

```swift
// Early in your app lifecycle, like in your AppDelegate:
LottieLogger.shared = .printToConsole // print warnings to console instead of calling Swift.assertionFailure
```

**Reply by calda** [reply]
To be fair, since this is a new assertion in 3.4.0 (missing images would fail silently in 3.3.0) its possible that this assertion is too aggressive and should be using `LottieLogger.shared.warn` (always just prints to console) instead of `LottieLogger.shared.assertionFailure`. An assertion seems right to me since this could cause parts of the animation to be missing. If you have a use case where it specifically makes sense for images to not be loaded, I'd be interested in hearing more.

**Reply by rfigini** [reply]
In our case, we don't have an image at that moment, because we replace it programatically. That's why "img_1.png" is not found, it's a dummy image name. Can We do something else in this case?

**Reply by calda** [reply]
Ok, that use case makes sense to me -- it seems like we should probably use `LottieLogger.shared.warn` instead of `LottieLogger.shared.assertionFailure` here. Would you be interested in making a pull request with that change?

**Reply by rfigini** [reply]
Yes for sure, I'll send you the pull request with this change adding a reference to this discussion in the description.

**Reply by rfigini** [reply]
Here's the [PR](https://github.com/airbnb/lottie-ios/pull/1680), thanks for your help.

**Reply by calda** [reply]
Thanks!
